osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineBlueprintContainer.cs

```diff
@@ -43,6 +51,29 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
             DragBox.Alpha = 0;
+
+            placement = beatmap.PlacementObject.GetBoundCopy();
+            placement.ValueChanged += placementChanged;
+        }
+
+        private void placementChanged(ValueChangedEvent<HitObject> obj)
+        {
+            if (obj.NewValue == null)
+            {
+                if (placementBlueprint != null)
+                {
+                    SelectionBlueprints.Remove(placementBlueprint);
+                    placementBlueprint = null;
+                }
+            }
+            else
+            {
+                placementBlueprint = CreateBlueprintFor(obj.NewValue);
+
+                placementBlueprint.Colour = Color4.MediumPurple;
+
+                SelectionBlueprints.Add(placementBlueprint);
```

Comment:
So what actually removes the placement from the selection blueprints in this case? I see the one above, which removes it if it's been set to null, but what about when the user cycles through their tools?

Reply:
on tool change placement is ended with commit=false